sdks/python/apache_beam/io/gcp/datastore/v1new/datastoreio.py

```diff
@@ -391,6 +392,7 @@ def add_element_to_batch(self, element):
         raise ValueError('Entities to be written to Cloud Datastore must '
                          'have complete keys:\n%s' % client_entity)
       self._batch.put(client_entity)
+      self._batch_mutations.append((helper._WRITE_MUTATION, client_entity))
```

Comment:
We could also pass it only once as another param to write_mutations, and by adding another property to each subclass.

Reply:
I think that moving write_mutations into DatastoreMutateFn will let you use add_element_to_batch instead of pairing a mutation_type with each entity (or passing it once).